src/main/java/uk/gov/hmcts/reform/pip/subscription/management/service/SubscriptionServiceImpl.java

```diff
@@ -96,4 +106,48 @@ UserSubscription collectSubscriptions(List<Subscription> subscriptions) {
         });
         return userSubscription;
     }
+
+    @Async
+    @Override
+    public void collectSubscribers(Artefact artefact) {
+        List<Subscription> subscriptionList = new ArrayList<>(querySubscriptionValue(
+            SearchType.COURT_ID.name(), artefact.getCourtId()));
+        artefact.getSearch().get("cases").forEach(object -> subscriptionList.addAll(extractSearchValue(object)));
+
+        List<Subscription> subscriptionsToEmail;
+        if (artefact.getSensitivity().equals(Sensitivity.CLASSIFIED)) {
+            subscriptionsToEmail = validateSubscriptionPermissions(subscriptionList, artefact.getListType());
+        } else {
+            subscriptionsToEmail = subscriptionList;
+        }
+
+        log.info("Subscriber list created. Notifying {} subscribers.", subscriptionsToEmail.size());
+    }
+
+    private List<Subscription> validateSubscriptionPermissions(List<Subscription> subscriptions, ListType listType) {
+        List<Subscription> filteredList = new ArrayList<>();
+        subscriptions.forEach(subscription -> {
+            if (accountManagementService.isUserAuthenticated(subscription.getUserId(), listType)) {
+                filteredList.add(subscription);
+            }
+        });
+        return filteredList;
+    }
+
+    private List<Subscription> querySubscriptionValue(String term, String value) {
+        return repository.findSubscriptionsBySearchValue(term, value);
+    }
+
+    private List<Subscription> extractSearchValue(Object caseObject) {
+        List<Subscription> subscriptionList = new ArrayList<>();
+        try {
+            subscriptionList.addAll(querySubscriptionValue(SearchType.CASE_ID.name(),
+                                                           ((LinkedHashMap) caseObject).get("caseNumber").toString()));
+            subscriptionList.addAll(querySubscriptionValue(SearchType.CASE_URN.name(),
+                                                           ((LinkedHashMap) caseObject).get("caseUrn").toString()));
+        } catch (NullPointerException ex) {
```

Comment:
Is there any way we can avoid catching the NullPointer here with explicit checks?

Reply:
this was added as case items dont seem to be required but we dont want to break on them, so NPE is caught to stop application breaking and the case item is skipped over